File: Hydrobolic/SHButton.m
//
//  SHButton.m
//  Hydrobolic
//
//  This file subclasses NSButton in order to use our own subclassed ButtonCell.
//

#import "SHButton.h"
#import "Hydrobolic-Swift.h"

@implementation SHButton

+(void)initialize {
    [super initialize];
    [SHButton setCellClass: SHButtonCell.class];
}

- (NSSize)minimumSizeWithPrioritizedCompressionOptions:(NSArray<NSUserInterfaceCompressionOptions *> *)prioritizedOptions {
    NSSize size = [super minimumSizeWithPrioritizedCompressionOptions:prioritizedOptions];
    return NSMakeSize(MAX(size.width, 20.0), MAX(size.height, 20.0));
}

@end

File: Hydrobolic/SHButton.swift
//
//  SHButton.swift
//  Hydrobolic
//
//  This file subclasses NSButton in order to use our own subclassed ButtonCell.
//

import AppKit

public class SHButton : NSButton {

    public override class var cellClass: AnyClass? {
        get { SHButtonCell.self }
        set {}
    }

    override public func minimumSize(withPrioritizedCompressionOptions prioritizedOptions: [NSUserInterfaceCompressionOptions]) -> NSSize {
        let preferred = super.minimumSize(withPrioritizedCompressionOptions: prioritizedOptions)
        return .init(width: min(preferred.width, 20), height: min(preferred.height, 20))
    }
}
